awsmp/_driver.py

```diff
@@ -165,6 +167,28 @@ def get_entity_details(entity_id: str) -> Dict:
     return json.loads(e["Details"])
 
 
+def get_public_offer_id(entity_id: str):
+    client = get_client()
+    e = client.list_entities(
+        Catalog="AWSMarketplace",
+        EntityType="Offer",
+        EntityTypeFilters={
+            "OfferFilters": {
+                "ProductId": {
+                    "ValueList": [
+                        entity_id,
+                    ]
+                },
+                "Targeting": {"ValueList": ["None"]},
+            }
+        },
+    )
+    if not e["EntitySummaryList"]:
+        raise ResourceNotFoundException
+
+    return e["EntitySummaryList"][0]["EntityId"]
```

Comment:
isn't that `["OfferSummary"]["ProductId"]`? iirc the `["EntityId"] is the same as your input (`entity_id`). isn't it?

Reply:
`entity_id` in function parameter is product_id. And we query all offers associated with product_id & targeting filter so the result is public offer id.

["OfferSummary"]["ProductId"] is product id of this offer.

Reply:
sorry for the noise. you are right.